### PR TITLE
JAMES-2441 polish guice

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please (build result lost)

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ResilientClusterProvider.java [hunk 1]
```diff
@@ -20,85 +20,53 @@
...
+    private ResilientClusterProvider(ClusterConfiguration configuration, AsyncRetryExecutor executor) throws ExecutionException, InterruptedException {
+        cluster = RetryExecutorUtil.retryOnExceptions(executor, configuration.getMaxRetry(), configuration.getMinDelay(), NoHostAvailableException.class)
+                .getWithRetry(getClusterRetryCallable(configuration))
                 .get();
     }
 
-    private RetryCallable<Cluster> getClusterRetryCallable(PropertiesConfiguration configuration, List<Host> servers, QueryLoggerConfiguration queryLoggerConfiguration) {
-        LOGGER.info("Trying to connect to Cassandra service at {} (list {})", LocalDateTime.now(), servers);
+    private RetryCallable<Cluster> getClusterRetryCallable(ClusterConfiguration configuration) {
+        LOGGER.info("Trying to connect to Cassandra service at {} (list {})", LocalDateTime.now(), configuration.getHosts());
```
Comment (rouazana) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ResilientClusterProvider.java]: implicit toString?

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

**Comment by aduprat** [comment]
Build started sha1 is merged.

**Comment by aduprat** [comment]
Build finished. No test results found.

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

**Comment by aduprat** [comment]
Build started sha1 is merged.

**Comment by aduprat** [comment]
Build finished. No test results found.

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

**Comment by aduprat** [comment]
Build started sha1 is merged.

**Comment by aduprat** [comment]
Build finished. No test results found.

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

**Comment by aduprat** [comment]
Build started sha1 is merged.

**Comment by aduprat** [comment]
Build finished. No test results found.

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

**Comment by aduprat** [comment]
Build started sha1 is merged.

**Comment by aduprat** [comment]
Build finished. No test results found.

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ResilientClusterProvider.java (hunk 1 above)
Comment (chibenwa) [reply server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ResilientClusterProvider.java]: Was there before.

Anyway I'll fix it.

File: server/container/guice/protocols/smtp/src/main/java/org/apache/james/modules/protocols/SMTPServerModule.java [hunk 2]
```diff
@@ -37,6 +37,8 @@
 
     @Override
     protected void configure() {
+        install(new JSPFModule());
```
Comment (chibenwa) [inline server/container/guice/protocols/smtp/src/main/java/org/apache/james/modules/protocols/SMTPServerModule.java]: This commit allows using JSPF hooks.

File: server/container/guice/jpa-common-guice/pom.xml [hunk 3]
```diff
@@ -46,6 +46,11 @@
             <groupId>${project.groupId}</groupId>
             <artifactId>james-server-guice-common</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.derby</groupId>
+            <artifactId>derby</artifactId>
+            <scope>test</scope>
```
Comment (aduprat) [inline server/container/guice/jpa-common-guice/pom.xml]: The bad point on this is that the `maven-dependency-plugin` plugin (which is copying the jars in order to have a lib folder for run time) is also copying the tests dependencies ;)

**Comment by aduprat** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**